module/zfs/zfs_vfsops.c

```diff
@@ -1701,8 +1701,7 @@ zfs_resume_fs(zfs_sb_t *zsb, const char *osname)
 		 * Since we couldn't setup the sa framework, try to force
 		 * unmount this file system.
 		 */
-		if (zsb->z_os)
-			(void) zfs_umount(zsb->z_sb);
+		(void) zfs_umount(zsb->z_sb);
```

Comment:
@behlendorf 
The commit message says concurrent umount, so it's likely caused by race. Though I don't think the patch actually fix the race. Also, using zfs_umount here looks strange, I'm not sure how things would work in terms of VFS after this call.